docs/mini-apps/features/context.mdx

```diff
@@ -0,0 +1,390 @@
+  }
+}
+```
+
+<AccordionGroup>
+<Accordion title="User Object Properties">
+- **fid**: Unique Farcaster identifier for the user
+- **username**: Handle without @ symbol  
+- **displayName**: User's chosen display name
+- **pfpUrl**: Profile picture URL
+</Accordion>
+</AccordionGroup>
+
+### Client Object
+
+Information about the host platform and device:
+
+```typescript Client Object Structure
+{
+  "client": {
+    "clientFid": 309857,        // Base app identifier
+    "added": false,             // App save status
+    "platformType": "mobile",   // Device type
+    "safeAreaInsets": {
+      "top": 0,
+      "bottom": 34,
+      "left": 0, 
+      "right": 0
+    }
+  }
+}
+```
+
+<AccordionGroup>
+<Accordion title="Client Object Properties">
+- **clientFid**: Identifies the host app (309857 = Base app)
+- **added**: Whether user has saved your mini app
+- **platformType**: Device type for responsive design
+- **safeAreaInsets**: Safe zones to avoid system UI overlap
+</Accordion>
+</AccordionGroup>
+
+### Location Object
+
+Describes how the user discovered your mini app - critical for viral optimization:
+
+<Tabs>
+<Tab title="Launcher Entry">
+```typescript
+{
+  "location": {
+    "type": "launcher"
+  }
+}
+```
+
+<Info>
+**User Journey**: Returning user opening from saved apps  
+**Strategy**: Focus on retention and progression features
+</Info>
+</Tab>
+
+<Tab title="Messaging Entry">
+```typescript
+{
+  "location": {
+    "type": "messaging"
+  }
+}
+```
+
+<Info>
+**User Journey**: Discovery through private messages  
+**Strategy**: Encourage broader social sharing
+</Info>
+</Tab>
+
+<Tab title="Cast Embed Entry">
+```typescript
+{
+  "location": {
+    "type": "cast_embed",
+    "embed": "https://frames-v2-demo-lilac.vercel.app/",
+    "cast": {
+      "author": {
+        "fid": 20390,
+        "username": "sohey",
+        "displayName": "Soheybuildson.base", 
+        "pfpUrl": "https://imagedelivery.net/..."
+      },
+      "fid": 20390,
+      "hash": "0x96178d931158cebbdad1b647b6549f0bb05709ed",
+      "timestamp": 1755781905000,
+      "text": "Check out this mini app!",
+      "embeds": ["https://frames-v2-demo-lilac.vercel.app/"]
+    }
+  }
+}
+```
+
+<Info>
+**User Journey**: Primary viral discovery through social feeds  
+**Strategy**: Optimize first-time experience and immediate engagement
+</Info>
+</Tab>
+</Tabs>
+
+## Implementation Guide
+
+<Steps>
+<Step title="Access Context Data">
+Import and use the MiniKit hook to access context:
+
+```typescript
+'use client';
+import { useMiniKit } from '@coinbase/onchainkit/minikit';
+import { useEffect } from 'react';
+
+function MyMiniApp() {
+  const { context, isFrameReady, setFrameReady } = useMiniKit();
+  
+  useEffect(() => {
+    if (!isFrameReady) {
+      setFrameReady();
+    }
+  }, [setFrameReady, isFrameReady]);
+  
+  // Always check for context availability
+  if (!context) {
+    return <div>Loading...</div>;
+  }
```

Comment:
issue: Doing this would make this mini app not work in the browser, since this will never resolve.

You could suggest using 

```ts
const { isInMiniApp } = useIsInMiniApp();

...

if (isInMiniApp && !context) {
  return <div>Loading...</div>;
}
```